### ISSUE TITLE
[Android] Return type allowed on separate line

### ISSUE THREAD

**Issue by camsteffen** [pinterest/ktlint#941]
Ktlint in Android mode allows the return type of a function signature to be on a separate line when the parameters are not on their own lines.

The Android Style Guide says ([link](https://developer.android.com/kotlin/style-guide#functions)):

> When a function signature does not fit on a single line, break each parameter declaration onto its own line. Parameters defined in this format should use a single indent (+4). The closing parenthesis ()) and return type are placed on their own line with no additional indent.

**It seems like ktlint does not consider the return type to be part of the function signature.**

## Example / Steps to Reproduce

```
ktlint --android --stdin <<< '
fun thisismytestmethod(aaaaaaaaaaaaaaaa: String, bbbbbbbbbbbbbb: String, ccccccccccccccc, String):
    String {
        return "test"
    }'
```

## Expected Behavior

Ktlint should detect that the function signature does not fit on one line and give error(s) that the parameters are not on their own line.

## Observed Behavior

No error.

## Your Environment

* Version of ktlint used: 0.39.0
* Operating System and version: macos 10.15.7

**Comment by camsteffen** [comment]
Actually, I think this applies to non-android usage as well. ([style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting))

**Comment by paul-dingemans** [comment]
> Actually, I think this applies to non-android usage as well. ([style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting))

I agree that this is not adroid specific.

With max-line-length set to 50, the generated output is as follows:
```
fun thisismytestmethod(
    aaaaaaaaaaaaaaaa: String,
    bbbbbbbbbbbbbb: String,
    ccccccccccccccc,
    String
):
    String {
    return "test"
}
```
In this case the parameter-list-wrapping rule forces the parameters to separate lines. Note however that the return type should be placed on the same line as the closing parentheses and colon. But also IntelliJ does not fix this mistake.

For example with max-line-length set to 100, the generated output is as follows:
```
fun thisismytestmethod(aaaaaaaaaaaaaaaa: String, bbbbbbbbbbbbbb: String, ccccccccccccccc, String):
    String {
    return "test"
}
```
Also the IntelliJ default formatting does not reformat this statement.

So the behavior of ktlint is consistent with IntelliJ default formatting. Technically it is not a bug in the current ktlint rules.

I am creating a new  rule `method-return-type-wrapping` to handle this and other edge cases regarding formatting of the method signature to format below:
```
fun thisismytestmethod(
    aaaaaaaaaaaaaaaa: String,
    bbbbbbbbbbbbbb: String,
    ccccccccccccccc: String
): String {
    return "test"
}
```

**Comment by paul-dingemans** [comment]
OP states that example below should not result in an error:
> ktlint --android --stdin <<< '
> fun thisismytestmethod(aaaaaaaaaaaaaaaa: String, bbbbbbbbbbbbbb: String, ccccccccccccccc, String):
>     String {
>         return "test"
>     }'

According to examples in [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#functions) and [Android's Kotlin Styleguide](https://developer.android.com/kotlin/style-guide#functions) the closing parenthesis and return type should be on the same line. This is also the style that is is used by rule `experimental:function-signature` which has been released in version `046.0`.

**Comment by camsteffen** [comment]
> OP states that example below should not result in an error:

Actually I said that it *should* result in an error. But yes, it looks like this is properly addressed with `experimental:function-signature`, thanks! Is this intended to become a default rule eventually?

**Comment by paul-dingemans** [comment]
Yes, at some point in time it will become standard. But it depends on the feedback that we receive about the rule.
